### ISSUE TITLE
[ww33] dEQP-VK.wsi.android.incremental_present have 8 tests failed on android.

### ISSUE THREAD

**Issue by jennycao** [intel/external-mesa#93]
dEQP-VK.wsi.android.incremental_present.scale_none.fifo#incremental_present | fail
dEQP-VK.wsi.android.incremental_present.scale_none.fifo#reference | fail
dEQP-VK.wsi.android.incremental_present.scale_none.mailbox#incremental_present | fail
dEQP-VK.wsi.android.incremental_present.scale_none.mailbox#reference | fail
dEQP-VK.wsi.android.incremental_present.scale_up.fifo#incremental_present | fail
dEQP-VK.wsi.android.incremental_present.scale_up.fifo#reference | fail
dEQP-VK.wsi.android.incremental_present.scale_up.mailbox#incremental_present | fail
dEQP-VK.wsi.android.incremental_present.scale_up.mailbox#reference | fail

**Comment by strassek** [comment]
It looks like all the subtests are failing in anv_GetSwapchainGrallocUsageANDROID with VK_IMAGE_USAGE_STORAGE_BIT unsupported. Some are failing early in anv_GetPhysicalDeviceImageFormatProperties2 with unsupported format, specifically for VK_FORMAT_R8G8B8A8_SRGB and VK_FORMAT_R5G6B5_UNORM_PACK16.

**Comment by tpalli** [comment]
As a hack you could "imageUsage &= ~VK_IMAGE_USAGE_STORAGE_BIT" in GetSwapChainGrallocUsageANDROID .. this should make some tests pass. Tests failing in GetPhysicalDeviceImageFormatProperties2 will need some investigation if there are some specific flags given why those formats would not work .. IMO they *should*.

**Comment by strassek** [comment]
These 2 hunks seem to be all that is needed to get the tests passing:
```
diff --git a/src/intel/isl/isl_storage_image.c b/src/intel/isl/isl_storage_image.c
index e2bd4dd..7cafebb 100644
--- a/src/intel/isl/isl_storage_image.c
+++ b/src/intel/isl/isl_storage_image.c
@@ -69,6 +69,8 @@ isl_is_storage_image_format(enum isl_format format)
    case ISL_FORMAT_R16_SNORM:
    case ISL_FORMAT_R8_UNORM:
    case ISL_FORMAT_R8_SNORM:
+   case ISL_FORMAT_R8G8B8A8_UNORM_SRGB:
+   case ISL_FORMAT_B5G6R5_UNORM:
       return true;
    default:
       return false;
diff --git a/src/intel/vulkan/anv_android.c b/src/intel/vulkan/anv_android.c
index 46c41d5..5c41db3 100644
--- a/src/intel/vulkan/anv_android.c
+++ b/src/intel/vulkan/anv_android.c
@@ -268,19 +268,13 @@ VkResult anv_GetSwapchainGrallocUsageANDROID(
                        "inside %s", __func__);
    }
 
-   /* Reject STORAGE here to avoid complexity elsewhere. */
-   if (imageUsage & VK_IMAGE_USAGE_STORAGE_BIT) {
-      return vk_errorf(device->instance, device, VK_ERROR_FORMAT_NOT_SUPPORTED,
-                       "VK_IMAGE_USAGE_STORAGE_BIT unsupported for gralloc "
-                       "swapchain");
-   }
-
    if (unmask32(&imageUsage, VK_IMAGE_USAGE_TRANSFER_DST_BIT |
                              VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT))
       *grallocUsage |= GRALLOC_USAGE_HW_RENDER;
 
    if (unmask32(&imageUsage, VK_IMAGE_USAGE_TRANSFER_SRC_BIT |
                              VK_IMAGE_USAGE_SAMPLED_BIT |
+                             VK_IMAGE_USAGE_STORAGE_BIT |
                              VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT))
       *grallocUsage |= GRALLOC_USAGE_HW_TEXTURE;
 
```
I would like some more clarity on why storage was being rejected.

**Comment by renchenglei** [comment]
@strassek & @tpalli , above patch could fix all wsi.android.* failure. Will we have plan to get it merged on github?

**Comment by strassek** [comment]
@renchenglei We aren't quite there yet. The above changes seem to be causing or exposing failures in dEQP-VK.subgroups.arithmetic#subgroupinclusive*.

edit: The subgroups.arithmetic failures I saw may have been unrelated, will need to do another test run to be sure. In any case we still need some feedback on the feasibility of adding these formats.

**Comment by renchenglei** [comment]
@strassek I finished one round of CTS VK test on Android, and with above change all wsi.android.* cases could pass and no regression found. Now, there are only two cases fail(one is #90 and another is #89). #89 should be fixed with @tpalli's patch #96(Thanks @tpalli ).

**Comment by renchenglei** [comment]
@strassek as this won't cause regression, will we have plan to submit the changes?

**Comment by strassek** [comment]
No, we expose formats to Android that can't be used for storage, so the above changes would still be incorrect.

**Comment by strassek** [comment]
@jennycao This should now be fixed in master, please retest.

**Comment by jennycao** [comment]
@strassek this issue is fixed by #100, we retest and got pass, Thanks for the fix :)
